string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,34 @@
+/* exported titleCase */
+function titleCase(title) {
+  let arr = title.toLowerCase();
+  arr = arr.split(' ');
+  for (var i = 0; i < arr.length; i++) {
+    if (arr[i] === 'the' && i === 0) {
+      arr[i] = arr[i][0].toUpperCase() + arr[i].slice(1);
+    } else if (i !== 0 && arr[i - 1].includes(':')) {
+      arr[i] = arr[i][0].toUpperCase() + arr[i].slice(1);
+    } else if (arr[i] === 'the' && i !== 0) {
+      arr[i].toLowerCase();
+    } else if (arr[i] === 'for') {
+      arr[i].toLowerCase();
+    } else if (arr[i] === 'self-taught') {
+      arr[i] = arr[i][0].toUpperCase() + arr[i].slice(1, 5) + arr[i][5].toUpperCase() + arr[i].slice(6);
+    } else if (i !== 0 && arr[i].includes('-')) {
+      if (arr[i][2] === '-') {
+        arr[i] = arr[i][0].toUpperCase() + arr[i].slice(1, 3) + arr[i][3].toUpperCase() + arr[i].slice(4);
```

Comment:
You cannot safely assume the length of a string just because it has a hyphen in it.